tests/ArmTemplates.Tests/Extractor/Scenarios/ParametersExtractorTests.cs

```diff
@@ -432,5 +433,63 @@ public void CreateResourceTemplateParameterTemplate_ProperlyGeneratesTemplate_Wi
         }
+
+        [Fact]
+        public async Task CreateResourceTemplateParameterTemplate_ProperlyGeneratesTemplate_WithBackendProxySettings()
+        {
+            // arrange
+            var currentTestDirectory = Path.Combine(this.OutputDirectory, nameof(CreateResourceTemplateParameterTemplate_ProperlyGeneratesTemplate_WithBackendProxySettings));
+
+            var extractorConfig = this.GetDefaultExtractorConsoleAppConfiguration(paramBackend: "true");
+            var extractorParameters = new ExtractorParameters(extractorConfig);
+            var extractorExecutor = this.GetExtractorInstance(extractorParameters, null);
+
+            var backends = new BackendTemplateResources()
+            {
+                BackendNameParametersCache = new Dictionary<string, BackendApiParameters>()
+                {
+                    { "key1", new BackendApiParameters() { Protocol = "protocol", ResourceId = "resourceId", Url = "url" } }
+                },
+                BackendProxyParametersCache = new Dictionary<string, BackendProxyParameters>()
+                {
+                    { "key1", new BackendProxyParameters() { Username = "username", Url = "url" } }
+                },
+                Backends = new List<BackendTemplateResource>()
+                {
+                    {
+                        new BackendTemplateResource
+                        {
+                            Name = "test"
+                        }
+                    }
+                }
+            };
+
+            // act
+            var parametersTemplate = await extractorExecutor.GenerateParametersTemplateAsync(null, null, backends, null, new IdentityProviderResources(), new OpenIdConnectProviderResources(), currentTestDirectory);
+
+            File.Exists(Path.Combine(currentTestDirectory, extractorParameters.FileNames.Parameters)).Should().BeTrue();
+
+            parametersTemplate.Parameters.Should().ContainKey(ParameterNames.BackendProxy);
+            parametersTemplate.Parameters.Should().ContainKey(ParameterNames.BackendSettings);
+
+            var backendProxyParameters = (TemplateObjectParameterProperties)parametersTemplate.Parameters[ParameterNames.BackendProxy];
+            var backendProxyParameterValues = (Dictionary<string, BackendProxyParameters>)backendProxyParameters.Value;
+
+            backendProxyParameterValues.Should().NotBeNull();
+            backendProxyParameterValues.ContainsKey("key1").Should().BeTrue();
+            backendProxyParameterValues["key1"].Username.Should().Be("username");
+            backendProxyParameterValues["key1"].Url.Should().Be("url");
+
+
+            var backendSettingsParameters = (TemplateObjectParameterProperties)parametersTemplate.Parameters[ParameterNames.BackendSettings];
+            var backendSettingsParameterValues = (Dictionary<string, BackendApiParameters>)backendSettingsParameters.Value;
+
+            backendSettingsParameterValues.Should().NotBeNull();
```

Comment:
this check is not checking anything actually - if the casting fails before, then it would already throw an [InvalidCastException](https://docs.microsoft.com/en-us/dotnet/api/system.invalidcastexception?view=net-6.0). So you can either delete this check or do code below, because `as` expression will not throw even if cast has failed 
```suggestion
var backendSettingsParameters = parametersTemplate.Parameters[ParameterNames.BackendSettings] as TemplateObjectParameterProperties;
            var backendSettingsParameterValues = backendSettingsParameters.Value as Dictionary<string, BackendApiParameters>;

            backendSettingsParameterValues.Should().NotBeNull();
```

Reply:
Could you please clarify why it should fail while casting reference object (in case it is null) which is also valid for Dictionary and TemplateObjectParameterProperties?

Reply:
nevermind - I somehow thought it will be a cast exception, if null would be casted. Silly me.